app/assets/javascripts/discourse/app/components/glimmer-site-header.gjs

```diff
@@ -0,0 +1,518 @@
+import Component from "@glimmer/component";
+import { DEBUG } from "@glimmer/env";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import { cancel, schedule } from "@ember/runloop";
+import { inject as service } from "@ember/service";
+import { waitForPromise } from "@ember/test-waiters";
+import ItsATrap from "@discourse/itsatrap";
+import concatClass from "discourse/helpers/concat-class";
+import scrollLock from "discourse/lib/scroll-lock";
+import SwipeEvents from "discourse/lib/swipe-events";
+import { isTesting } from "discourse-common/config/environment";
+import discourseLater from "discourse-common/lib/later";
+import { bind, debounce } from "discourse-common/utils/decorators";
+import GlimmerHeader from "./glimmer-header";
+
+let _menuPanelClassesToForceDropdown = [];
+const PANEL_WIDTH = 340;
+
+export default class GlimmerSiteHeader extends Component {
+  @service appEvents;
+  @service currentUser;
+  @service site;
+  @service header;
+
+  @tracked headerWrap = null;
+  @tracked pxClosed = null;
+  @tracked headerElement = null;
+  docking;
+  @tracked _dockedHeader = false;
+  @tracked _swipeMenuOrigin = null;
+  @tracked _swipeEvents = null;
+  @tracked _applicationElement = null;
+  @tracked _resizeObserver = null;
+  @tracked _docAt = null;
+  @tracked _animate = false;
+
+  constructor() {
+    super(...arguments);
+    this.docking = new Docking(this.dockCheck);
+
+    if (this.currentUser?.staff) {
+      document.body.classList.add("staff");
+    }
+
+    schedule("afterRender", () => this.animateMenu());
+  }
+  get dropDownHeaderEnabled() {
+    return !this.sidebarEnabled || this.site.narrowDesktopView;
+  }
+
+  get leftMenuClass() {
+    if (document.querySelector("html").classList["direction"] === "rtl") {
```

Comment:
`classList["direction"]` again.

(plus, btw, `document.querySelector("html")` can be replaced w/ `document.documentElement`)